worker/src/sql.rs

```diff
@@ -213,4 +213,43 @@ impl SqlCursor {
     pub fn rows_written(&self) -> usize {
         self.inner.rows_written() as usize
     }
+
+    /// JavaScript Iterator.next() implementation.
+    ///
+    /// Returns an object with `{ done, value }` properties. This allows the cursor
+    /// to be used as a JavaScript iterator.
+    pub fn next(&self) -> js_sys::Object {
+        self.inner.next()
+    }
+
+    /// Returns an iterator where each row is an array rather than an object.
+    ///
+    /// This method provides a more efficient way to iterate over results when you
+    /// only need the raw values without column names.
+    pub fn raw(&self) -> js_sys::Iterator {
```

Comment:
Rather than having these return `js_sys` types I think it would be good to have these written as first-class Rust APIs, so `next()` with a signature more like the Rust iterator case returning `Option<T>` and `raw()` returning `Option<Vec<SqlStorageValue>>` I believe.